docs/src/main/sphinx/security/inspect-jks.rst

```diff
@@ -0,0 +1,148 @@
+.. _jks_certs:
+
+=========
+JKS files
+=========
+
+This topic describes the :ref:`Java KeyStore (JKS) <glossJKS>` certificate
+container formats accepted by Trino to validate TLS/HTTPS client access to the
+cluster.
+
+.. _server_java_keystore:
+
+Java keystore
+-------------
+
+.. note::
+
+  See :ref:`Inspect PEM files <pem_certs>` for information on the preferred
+  certificate format.
+
+The Java KeyStore (JKS) system is provided as part of your Java installation.
+Private keys and certificates for your server are stored in a *keystore* file.
+The keystore file is always password-protected.
+
+To be recognized as valid by the Trino :doc:`CLI </installation/cli>` and modern
+browsers, the key must be signed by a trusted Certificate Authority (CA). For
+JKS keys, this means either that a certificate for the key's signing authority
+is listed in the current JVM's :ref:`truststore <cli_java_truststore>`, or that
+the certificate received from a CA is chained all the way back to a CA in the
+truststore.
+
+If you receive a keystore file from your network group or from a vendor,
+inspect the keystore's readiness to work with Trino. Then :ref:`configure
+<configure-https>` the server to accept client connections using HTTPS.
+
+.. _troubleshooting_keystore:
+
+Inspect and validate keystore
+-----------------------------
+
+Inspect the keystore file to make sure it contains the correct information for
+your Trino server. Use the ``keytool`` command, which is installed as part of
+your Java installation, to retrieve information from your keystore container
+file:
+
+.. code-block:: text
+
+  keytool -list -v -keystore yourKeystore.jks
+
+Keystores always require a password. If not provided on the ``keytool`` command
+line, ``keytool`` prompts for the password.
+
+Independent of the keystore's password, it is possible that the JKS key has its
+own password. It is easiest to make sure these passwords are the same. If the
+JKS key inside the keystore has a different password, you can be prompted twice.
+
+In the output of the ``keytool -list`` command, look for:
+
+* The keystore must contain a private key, such as ``Entry type:
+  PrivateKeyEntry``
+* Depending on the origin of your keystore file, it may also contain a
+  certificate. Example: ``Entry type: trustedCertEntry``
+* Confirm that the ``Valid from ... until`` entry shows no more than 398 days.
+* Verify that the Subject Alternative Name matches your server's DNS name.
+
+  Example:
+
+  .. code-block:: text
+
+    SubjectAlternativeName [
+        DNSName:  cluster.example.com
+    ]
+
+The JKS system works in conjunction with a truststore, described next.
+
+.. _import_to_keystore:
+
+Add PEM to keystore
```

Comment:
Drop this.  If you have a PEM, just use the PEM

Reply:
I began this section with "Your site may have standardized on using JKS semantics for all servers." 

Is that not a valid scenario to write for? There must be plenty of shops that are stuck in the truststore-only world.